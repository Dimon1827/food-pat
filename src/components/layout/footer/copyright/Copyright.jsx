import './Copyright.scss'

function Copyright() {
  return (
    <div className="copyright">
      <a className="copyright__link" href="#">
        <span className="copyright__text">Foo</span>
        <span className="copyright__text">die</span>
        <svg
          className="copyright__image"
          width="27"
          height="34"
          aria-hidden="true"
          focusable="false"
        >
          <use href="images/icons/stack.svg#partner"></use>
        </svg>
      </a>
    </div>
  );
}

export default Copyright;
