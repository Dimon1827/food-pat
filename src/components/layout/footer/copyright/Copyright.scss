@import '@/assets/styles/variables';

.copyright {
  @media (max-width: $mobile-width) {
    display: flex;
  }
}

.copyright__link {
  display: flex;
  align-items: center;
  margin: 0 auto;
  color: $primary-color;
  fill: $primary-color;
  text-decoration: none;

  &:hover {
    opacity: 0.8;
  }

  &:active {
    fill: $bg-basic-green;
    opacity: 0.3;
  }
}

.copyright__text {
  font-size: 20px;
  line-height: 125%;

  &:nth-child(1) {
    color: #f03328;
  }

  &:nth-child(2) {
    color: #ff9e0c;
  }


  @media (min-width: $tablet-width) {
    font-size: 24px;
  }
}

.copyright__image {
  margin-left: auto;

  @media (min-width: $tablet-width) {
    margin-left: 23px;
  }
}