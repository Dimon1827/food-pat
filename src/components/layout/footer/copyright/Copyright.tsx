import './Copyright.scss';

function Copyright() {
  return (
    <div className="copyright" data-testid="copyright">
      <div className="copyright__link">
        <span className="copyright__text">Foo</span>
        <span className="copyright__text">die</span>
        <svg
          className="copyright__image"
          width="27"
          height="34"
          aria-hidden="true"
          focusable="false"
        >
          <use href="images/icons/stack.svg#partner"></use>
        </svg>
      </div>
    </div>
  );
}

export default Copyright;
