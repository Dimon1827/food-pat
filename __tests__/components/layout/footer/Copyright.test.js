import Copyright from '@/components/layout/footer/copyright/Copyright';
import { screen, render } from '@testing-library/react';

describe('Copyright', () => {
  it('copyright renders correctly', () => {
    render(<Copyright />);
    expect(screen.getByTestId('copyright')).toMatchSnapshot();
  });
});
