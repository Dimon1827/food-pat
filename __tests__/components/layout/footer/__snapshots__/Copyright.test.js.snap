// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Copyright copyright renders correctly 1`] = `
<div
  class="copyright"
  data-testid="copyright"
>
  <div
    class="copyright__link"
  >
    <span
      class="copyright__text"
    >
      Foo
    </span>
    <span
      class="copyright__text"
    >
      die
    </span>
    <svg
      aria-hidden="true"
      class="copyright__image"
      focusable="false"
      height="34"
      width="27"
    >
      <use
        href="images/icons/stack.svg#partner"
      />
    </svg>
  </div>
</div>
`;
